locket/__init__.py

```diff
@@ -10,22 +10,35 @@
     import fcntl
 except ImportError:
     try:
+        import ctypes
+        import ctypes.wintypes
         import msvcrt
     except ImportError:
-        raise ImportError("Platform not supported (failed to import fcntl, msvcrt)")
+        raise ImportError("Platform not supported (failed to import fcntl, ctypes, msvcrt)")
     else:
+        _WinAPI_LockFile = ctypes.windll.kernel32.LockFile
```

Comment:
Use 

    kernel32 = ctypes.WinDLL('kernel32', use_last_error=True)

Avoid `windll` and `cdll`. The global `LibraryLoader` instances cache each loaded `WinDLL` or `CDLL` instance, which in turn caches function pointers that are accessed as attributes. Other projects that create a function pointer via `windll.kernel32` will conflict with your prototype (i.e. `restype`, `argtypes`, `errcheck`). This was formerly a problem for pyreadline and colorama. 

`windll` also doesn't allow specifying `use_last_error` to reliably capture the thread's last error value, which is available via `ctypes.get_last_error()`. Similarly, `cdll` doesn't allowing specifying `use_errno` to enable `ctypes.get_errno()`.